packages/clerk-js/src/ui/ImpersonationFab/ImpersonationFab.tsx

```diff
@@ -91,14 +95,63 @@ export const ImpersonationFab = () => {
   const session = useCoreSession();
   const { t } = useLocalizations();
   const { parsedInternalTheme } = useAppearance();
+  const ref = useRef<HTMLDivElement>(null);
   const actor = session?.actor;
   const isImpersonating = !!actor;
 
   //essentials for calcs
-  const top = '109px';
-  const right = '23px';
   const eyeWidth = parsedInternalTheme.sizes.$16;
   const eyeHeight = eyeWidth;
+  const topProperty = '--cl-impersonation-fab-top';
+  const rightProperty = '--cl-impersonation-fab-right';
+  const defaultTop = 109;
+  const defaultRight = 23;
+
+  const handleResize = () => {
+    const { current } = ref;
+    if (!current) {
+      return;
+    }
+
+    if (current.offsetLeft < 0 || current.offsetLeft > window.innerWidth) {
+      document.documentElement.style.setProperty(rightProperty, `${defaultRight}px`);
+    }
+  };
+
+  const onMouseDown: MouseEventHandler = () => {
+    window.addEventListener('mousemove', onMouseMove);
+    window.addEventListener(
+      'mouseup',
+      () => {
+        window.removeEventListener('mousemove', onMouseMove);
+        handleResize();
+      },
+      { once: true },
+    );
+  };
+
+  const onMouseMove = React.useCallback((e: MouseEvent) => {
+    e.stopPropagation();
+    e.preventDefault();
+    const { current } = ref;
+    if (!current) {
+      return;
+    }
+    document.documentElement.style.setProperty(
+      rightProperty,
+      `${window.innerWidth - current.offsetLeft - current.offsetWidth - e.movementX}px`,
+    );
+    document.documentElement.style.setProperty(topProperty, `${current.offsetTop + e.movementY}px`);
+  }, []);
+
+  //reposition the fab if needed when the window resizes
+  useEffect(() => {
+    window.addEventListener('resize', handleResize);
+
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  });
```

Comment:
🛑 I think we're missing an empty `[]` dependency array here
```suggestion
  }, []);
```

Reply:
Great tip!